api/v1/verticadb_webhook.go

```diff
@@ -200,7 +200,7 @@ func (v *VerticaDB) checkValidSubclusterTypeTransition(oldObj *VerticaDB, allErr
 		} else if oldType == SecondarySubcluster && sc.Type != SecondarySubcluster {
 			_, found := scToSbMap[sc.Name]
 			// You can only transition out of a secondary subcluster if its during sandboxing.
-			if sc.Type != SandboxPrimarySubcluster || !found {
+			if sc.IsSandboxPrimary(oldObj) || !found {
 				invalidStateTransitionErr(i)
```

Comment:
Now that there is a type under spec.sandboxes, we should no longer allow spec.subclusters.type to change.

Reply:
Updated to not allow subcluster type change if it's in a sandbox.